### ISSUE TITLE
[1.18.1][CoL] Tag forge:visual/water used before it was bound

### ISSUE THREAD

**Issue by drakray** [cech12/BucketLib#6]
Crash on load after adding BucketLib to pack
https://pastebin.com/0JnynTeW

```
Description: Initializing game

java.lang.IllegalStateException: Tag forge:visual/water used before it was bound
	at net.minecraft.tags.StaticTagHelper$Wrapper.m_13263_(StaticTagHelper.java:142) ~[client-1.18.1-20211210.034407-srg.jar%23120!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.tags.StaticTagHelper$Wrapper.m_8110_(StaticTagHelper.java:153) ~[client-1.18.1-20211210.034407-srg.jar%23120!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.material.Fluid.m_76108_(Fluid.java:95) ~[client-1.18.1-20211210.034407-srg.jar%23120!/:?] {re:mixin,re:classloading}
	at com.telepathicgrunt.the_bumblezone.fluids.SugarWaterFluid.m_6212_(SugarWaterFluid.java:149) ~[the_bumblezone_forge-4.3.6+1.18.1.jar%23113!/:4.3.6+1.18.1] {re:classloading}
	at cech12.bucketlib.api.item.UniversalBucketItem.m_6787_(UniversalBucketItem.java:411) ~[bucketlib-1.18.1-0.5.0.0.jar%2364!/:1.18.1-0.5.0.0] {re:classloading}
	at net.minecraft.client.Minecraft.m_91271_(Minecraft.java:711) ~[client-1.18.1-20211210.034407-srg.jar%23120!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architectury.mixins.json:MixinMinecraft,pl:mixin:APP:balm.mixins.json:MinecraftMixin,pl:mixin:APP:spyglass_improvements.mixins.json:MinecraftMixin,pl:mixin:APP:the_bumblezone.mixins.json:client.MinecraftMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:504) ~[client-1.18.1-20211210.034407-srg.jar%23120!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architectury.mixins.json:MixinMinecraft,pl:mixin:APP:balm.mixins.json:MinecraftMixin,pl:mixin:APP:spyglass_improvements.mixins.json:MinecraftMixin,pl:mixin:APP:the_bumblezone.mixins.json:client.MinecraftMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:169) ~[client-1.18.1-20211210.034407-srg.jar%23120!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:45) ~[fmlloader-1.18.1-39.1.1.jar%2323!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.1.3.jar%235!/:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:149) [bootstraplauncher-1.0.0.jar:?] {}

```

Minecraft 1.18.1
Forge 39.1.1
BucketLib 0.5.0.0
buncha others, listed in pastebin link

**Comment by cech12** [comment]
Thanks for your report, I will look into that :)

**Comment by cech12** [comment]
I think, I found the code that causes this issue. 

https://github.com/cech12/BucketLib/blob/30393ac5fed5d51f7693b2e654cb9ab175160d1b/src/main/java/cech12/bucketlib/api/item/UniversalBucketItem.java#L411

The check if a fluid is milk uses the `issame` method that is overridden by the Bumblezone mod. 

Until I fix that, you could use the BucketLib version 1.18.1-0.3.1.0.

**Comment by cech12** [comment]
Hopefully fixed and released in version 1.18.1-0.5.0.1.
If not, please let me know :)
